src/app/clusters/laundry-washer-controls-server/laundry-washer-controls-server.cpp

```diff
@@ -0,0 +1,231 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/util/af.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/config.h>
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/callback.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/enums.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandler.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/clusters/laundry-washer-controls-server/laundry-washer-controls-server.h>
+#include <app/server/Server.h>
+#include <app/util/error-mapping.h>
+#include <lib/core/CHIPEncoding.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::LaundryWasherControls;
+using namespace chip::app::Clusters::LaundryWasherControls::Attributes;
+using chip::Protocols::InteractionModel::Status;
+
+LaundryWasherControlsServer LaundryWasherControlsServer::sInstance;
+
+/**********************************************************
+ * LaundryWasherControlsServer public methods
+ *********************************************************/
+
+CHIP_ERROR LaundryWasherControlsServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    if (aPath.mClusterId != LaundryWasherControls::Id)
+    {
+        // We shouldn't have been called at all.
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    switch (aPath.mAttributeId)
+    {
+    case Attributes::SpinSpeeds::Id:
+        return ReadSpinSpeeds(aPath, aEncoder);
+    case Attributes::SupportedRinses::Id:
+        return ReadSupportedRinses(aPath, aEncoder);
+    default:
+        break;
```

Comment:
So the assumption here is that the application will directly write the SpinSpeedCurrent and NumberOfRinses into the attribute store when they change, and watch for writes to the attribute store to see when they are changed over the wire?

That's really not ideal; we should have a cluster API for the application to read/write these values and writes over the wire should notify the application via the delegate it will need to provide anyway to implement the lists.

Reply:
We have crated the API for the other 2 attributes, SpinSpeedCurrent and NumberOfRinses.
The writable setting of both SpinSpeeds and SupportedRinses are false. Should we add the API that they can be set? Since for reading the attribute are from the delegate, the value should be in the application.